### Vamos a filtrar el dataset para que solo muestre las 5 calles con más accidentes de madrid

In [176]:
# Paso 1: Identificar las 5 localizaciones con más accidentes
accidents_by_street = df.groupby('localizacion').size().reset_index(name='num_accidents')
top_streets = accidents_by_street.nlargest(5, 'num_accidents')

# Paso 2: Filtrar los datos originales para estas localizaciones
top_locations_data = df[df['localizacion'].isin(top_streets['localizacion'].tolist())]

# Paso 3: Agrupar por 'localizacion' y obtener el modo para 'tipo_accidente' y 'estado_meteorológico'
modes = top_locations_data.groupby('localizacion').agg({
    'tipo_accidente': lambda x: x.mode().iloc[0],  # Moda de 'tipo_accidente'
    'estado_meteorológico': lambda x: x.mode().iloc[0]  # Moda de 'estado_meteorológico'
}).reset_index()

# Paso 4: Unir esta información con el número total de accidentes para cada localización
top_accident_locations_with_modes = pd.merge(
    top_streets,
    modes,
    on='localizacion',
    how='left'
)
# Paso 5: Agregar las coordenadas representativas de cada localización
top_accident_locations_with_mode = pd.merge(
    top_accident_locations_with_modes,
    df[['localizacion', 'coordenada_x_utm', 'coordenada_y_utm']].drop_duplicates('localizacion'),
    on='localizacion',
    how='left'
)

top_accident_locations_with_mode

localizacion  num_accidents  \
0                                AUTOV. A-2, +00500E            172   
1                               AUTOV. A-42, +00500E            123   
2  CALL. PRINCIPE DE VERGARA / CALL. FRANCISCO SI...             94   
3                                     AUTOV. M-23, 0             78   
4                     AVDA. ARCENTALES / AUTOV. M-40             76   

            tipo_accidente estado_meteorológico coordenada_x_utm  \
0                  Alcance            Despejado       444511,644   
1        Colisión múltiple            Despejado       438952,303   
2  Colisión fronto-lateral            Despejado       442416,005   
3                  Alcance            Despejado        444922,63   
4                  Alcance            Despejado       449324,554   

  coordenada_y_utm  
0      4477569,316  
1      4469985,021  
2      4476670,396  
3      4474127,759  
4       4475697,19

Los detalles relevantes, son los siguientes:

**1. Localización: AUTOV. A-2, +00500E**
Número de Accidentes: 172
Tipo de Accidente más Frecuente: Alcance
Estado Meteorológico más Frecuente: Despejado
Coordenadas: X: 444511,644, Y: 4477569,316

**2. Localización: AUTOV. A-42, +00500E**
Número de Accidentes: 123
Tipo de Accidente más Frecuente: Colisión múltiple
Estado Meteorológico más Frecuente: Despejado
Coordenadas: X: 438952,303, Y: 4469985,021

**3.Localización: CALL. PRINCIPE DE VERGARA / CALL. FRANCISCO SILVELA**
Número de Accidentes: 94
Tipo de Accidente más Frecuente: Colisión fronto-lateral
Estado Meteorológico más Frecuente: Despejado
Coordenadas: X: 442416,005, Y: 4476670,396

**4. Localización: AUTOV. M-23, 0**
Número de Accidentes: 78
Tipo de Accidente más Frecuente: Alcance
Estado Meteorológico más Frecuente: Despejado
Coordenadas: X: 444922,63, Y: 4474127,759

**5. Localización: AVDA. ARCENTALES / AUTOV. M-40**
Número de Accidentes: 76
Tipo de Accidente más Frecuente: Alcance
Estado Meteorológico más Frecuente: Despejado
Coordenadas: X: 449324,554, Y: 4475697,19

#### Creamos un mapa para visualizar las calles con mayor Índice de accidentes en Madrid

In [1]:
from my_map_creator import MapCreator

mapa_top_loc = MapCreator(top_accident_locations_with_mode, lat_col_name="coordenada_x_utm",
                      long_col_name="coordenada_y_utm")

mapa_top_loc.set_columns_for_label(['localizacion', 'num_accidents', 'tipo_accidente', 'estado_meteorológico'])
mapa_top_loc.create_map(map_name="mapa_top_loc")

ModuleNotFoundError: No module named 'selenium'

In [ ]:
visualizador_accidentes(df.copy(), 'AUTOV. A-2, +00500E')

In [ ]:
display_img('AUTOV. A-2, +00500E')